cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -175,20 +175,19 @@ std::set<StreamId> LocalVersionedEngine::list_streams_internal(
     std::optional<SnapshotId> snap_name,
     const std::optional<std::string>& regex,
     const std::optional<std::string>& prefix,
-    const std::optional<bool>& opt_use_symbol_list,
-    const std::optional<bool>& opt_all_symbols
+    const std::optional<bool>& use_symbol_list,
```

Comment:
Since we don't use `value_or` does it make sense for this to be optional anymore? `use_symbol_list.value_or(cfg().symbol_list())` will throw if there is no value so we're kind of expecting there to be one. Why not enforce it in the definition of the function?